astropy/tests/tests/test_runner.py

```diff
@@ -42,11 +46,10 @@ class Spam(_TestRunnerBase):
         def spam(self, spam, kwargs):
             return [spam]
 
-    r = Spam(".")
-
-    args = r._generate_args(spam="spam")
-
-    assert ["spam"] == args
+    with pytest.raises(AstropyDeprecationWarning):
+        r = Spam(".")
+        args = r._generate_args(spam="spam")
+        assert ["spam"] == args
```

Comment:
While we're at it, maybe we could also enforce the more common idiom `assert results == expected` ?
```suggestion
        assert args == ["spam"]
```